bin/vcs-jump

```diff
@@ -103,8 +103,8 @@ def mode_diff(args)
 
   diff.lines.each do |line|
     # setting the inner Perl hacker free since 2007
-    (line =~ %r{^\+\+\+ b/(.*)}) ? (file = root + $~[1]) : (next unless file)
-    (line =~ %r{^@@ .*\+(\d+)}) ? (idx = $~[1].to_i) : (next unless idx)
+    (line =~ %r{^\+\+\+ (.*)}) ? (file = root + $~[1]) : (next unless file)
+    (line =~ %r{^@@ .*?\+(\d+)}) ? (idx = $~[1].to_i) : (next unless idx)
```

Comment:
So here's the original Perl regexen:

```perl
	if (m{^\+\+\+ (.*)}) { $file = $1; next }
	defined($file) or next;
	if (m/^@@ .*?\+(\d+)/) { $line = $1; next }
```

Thoughts on the first line: this change is seems broken. Note that it only works because the upstream script invokes `git diff` with `--no-prefix`. ie. instead of producing a "standard" diff with headers that look like this:

```diff
diff --git a/aspects/dotfiles/files/.zsh/zsh-syntax-highlighting b/aspects/dotfiles/files/.zsh/zsh-syntax-highlighting
--- a/aspects/dotfiles/files/.zsh/zsh-syntax-highlighting
+++ b/aspects/dotfiles/files/.zsh/zsh-syntax-highlighting
@@ -1 +1 @@
-Subproject commit 82cf2527fc13fefee6e87f99171c1b368f0abdb9
+Subproject commit 82cf2527fc13fefee6e87f99171c1b368f0abdb9-dirty
```
It produces:

```diff
diff --git aspects/dotfiles/files/.zsh/zsh-syntax-highlighting aspects/dotfiles/files/.zsh/zsh-syntax-highlighting
--- aspects/dotfiles/files/.zsh/zsh-syntax-highlighting
+++ aspects/dotfiles/files/.zsh/zsh-syntax-highlighting
@@ -1 +1 @@
-Subproject commit 82cf2527fc13fefee6e87f99171c1b368f0abdb9
+Subproject commit 82cf2527fc13fefee6e87f99171c1b368f0abdb9-dirty
```

You'll note that `vcs-jump` instead invokes `git diff` with `git -c diff.mnemonicPrefix=no diff`. From `man git-config`:

```
       diff.mnemonicPrefix
           If set, git diff uses a prefix pair that is different from the standard "a/" and "b/" depending on what is being compared. When
           this configuration is in effect, reverse diff output also swaps the order of the prefixes:
```

Looks like they used to do a similar thing to `vcs-jump` but changed it in https://github.com/git/git/commit/6108b04b708f6aa5d4a9349a240b5261fdab0cdf.

So that means that with this change, and depending on local config values, the jump list will be populated with entries like `a/some/file.txt` instead of the desired `some/file.txt`.

Relevant bits from `man git-diff`:

```
       --src-prefix=<prefix>
           Show the given source prefix instead of "a/".

       --dst-prefix=<prefix>
           Show the given destination prefix instead of "b/".

       --no-prefix
           Do not show any source or destination prefix.
```

Note also that this means that they used to jump to the post-image file-name (which I think is better, because it will land on the renamed file if rename detection is active), but now they jump to the pre-image filename.

I think this explains why it didn't work for you before but it works with out-of-the-box settings. There is likely something in you `git config --list` output that would account for the behavior difference.

So, I don't think we should apply this change. Instead, we should pass additional config options to force the use of prefixes.

```
git -c diff.noprefix=no -c diff.mnemonicPrefix=no
```

Thoughts on the second line: this is just turning it into a non-greedy pattern. They made the change [here](https://github.com/git/git/commit/74a7fa44d36c3e93febccdea5f44ff78555463d0). I expect this is an exceedingly rare edge case and doesn't explain the problem you were having, but we should apply the change anyway.